### PR TITLE
add AdaptiveAvgPooling2D

### PR DESCRIPTION
**Pull request by migeed-z**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61310
* #61265
* **#61239**
* #61238
* #61188

Differential Revision: [D29626359](https://our.internmc.facebook.com/intern/diff/D29626359)

### CODE REVIEW

File: torch/fx/experimental/graph_gradual_typechecker.py [hunk 1]
```diff
@@ -377,6 +377,61 @@ def linear_inference_rule(n: Node, op_type):
...
+        new_type_list[-1] = output_size[1]
+        new_type_list[-2] = output_size[0]
+
+        return TensorType(tuple(new_type_list))
+
+    else:
+        raise TypeError(f'Tensor ranks must be 3 or 4. Got {tensor_type}')
+
+@register_inference_rule(torch.nn.AdaptiveAvgPool2d)
+def adaptiveavgpool2d_inference_rule(n: Node, op_type):
```
Comment (jamesr66a) [inline torch/fx/experimental/graph_gradual_typechecker.py:404]: Rename `op_type`

**Review by jamesr66a — APPROVED** [review]
Cool!

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29626359).

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29626359).

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29626359).

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29626359).

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29626359).

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29626359).

**Pull request closed without merging**